src/Chapter2.hs

```diff
@@ -841,7 +875,13 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l = if n < 0
+             then []
+             else let len = length l
+                      lc = cycle l
+                      m = n `mod` len
+                   in take len $ drop m lc
```

Comment:
Nice implementation! Note, that `cycle` fails on empty lists, and `mod` fails with "division by zero" error when `length` is zero. But due to Haskell laziness and the implementation of `take`, you don't see such errors. You can write simple code, and it works on all cases automatically :slightly_smiling_face: